sdk/formrecognizer/Azure.AI.FormRecognizer/src/BoundingRegion.cs

```diff
@@ -37,8 +40,84 @@ internal BoundingRegion(int pageNumber, IReadOnlyList<PointF> boundingPolygon)
 
         /// <summary>
         /// The bounding polygon that outlines this region. Units are in pixels for images and inches for
-        /// PDF. The <see cref="LengthUnit"/> type of a recognized page can be found at <see cref="DocumentPage.Unit"/>.
+        /// PDF. The <see cref="DocumentPageLengthUnit"/> type of a recognized page can be found at <see cref="DocumentPage.Unit"/>.
         /// </summary>
         public IReadOnlyList<PointF> BoundingPolygon { get; }
+
+        /// <summary>
+        /// Indicates whether the current <see cref="BoundingRegion"/> is equal to another object of the same type.
+        /// They are considered equal if they have the same type, the same <see cref="PageNumber"/>, and the same point
+        /// coordinates in <see cref="BoundingPolygon"/>.
+        /// </summary>
+        /// <param name="obj">An object to compare with this <see cref="BoundingRegion"/>.</param>
+        /// <returns><c>true</c> if the current <see cref="BoundingRegion"/> is equal to the <paramref name="obj"/> parameter; otherwise, <c>false</c>.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => obj is BoundingRegion other && Equals(other);
+
+        /// <summary>
+        /// Indicates whether the current <see cref="BoundingRegion"/> is equal to another object of the same type.
+        /// They are considered equal if they have the same <see cref="PageNumber"/> and the same point coordinates
+        /// in <see cref="BoundingPolygon"/>.
+        /// </summary>
+        /// <param name="other">An object to compare with this <see cref="BoundingRegion"/>.</param>
+        /// <returns><c>true</c> if the current <see cref="BoundingRegion"/> is equal to the <paramref name="other"/> parameter; otherwise, <c>false</c>.</returns>
+        public bool Equals(BoundingRegion other)
+        {
+            if (PageNumber != other.PageNumber)
+            {
+                return false;
+            }
+
+            if (BoundingPolygon == null)
+            {
+                return other.BoundingPolygon == null;
+            }
+
+            if (BoundingPolygon.Count != other.BoundingPolygon.Count)
+            {
+                return false;
+            }
+
+            // Since points in a polygon are cyclical, two polygons could still be considered the same
+            // if points were shifted. For example, [(1,2),(3,4),(5,6),(7,8)] and [(7,8),(1,2),(3,4),(5,6)]
+            // represent the same polygon. This, however, is not expected to happen since points returned
```

Comment:
Does service guarantee this behavior?
Should we have a test to validate that the service is not breaking this since we are relying on this behavior.

Reply:
This service guarantees this behavior, according to their [documentation](https://github.com/Azure/azure-rest-api-specs/blob/2fd7dcd89afa70ff5ba7be88bee987da62099a28/specification/cognitiveservices/data-plane/FormRecognizer/stable/2022-08-31/FormRecognizer.json#L2063).
> Bounding polygon, with coordinates specified relative to the top-left of the page. The numbers represent the x, y values of the polygon vertices, clockwise from the left (-180 degrees inclusive) relative to the element orientation.

I think it would be easier to just update this method to not rely on this behavior anymore. Just need to check if points are shifted. I'll also add a test to ensure we're doing this correctly.

Reply:
Now that I thought more about it, it's not as simple since we would need to implement `GetHashCode` in a way that ensures shifted polygons generate the same hash code. I'll just add validation for the service then.

Reply:
Done. Had to update `GetHashCode` too, since shifted points should still give the same hash code. We're basically ordering the points before generating the code.